### PR TITLE
Research doc patch

### PR DESCRIPTION
**Pull request by kabilanmohanraj**
### Changes:
1. I've committed the research documents that I've been developing over the course of this summer to this repository. These documents elucidate the details about the various models I've read about and elucidate the criteria based on which the models were selected and analyzed.

2. Minor enhancements and updates have been made to the ReadMe file to reflect the recent changes and additions more accurately.

3. Add Dockerfile and instructions to build the image and mount folders from the host to the container.

### CODE REVIEW

File: trocr/evaluation-dataset/handwritten-typed-text-classification/research.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+#### Methodology
+Given the encountered limitations with CNNs, I approached the classification task in two primary steps to circumvent the challenges:
+
+1. **Feature Extraction with TrOCR Encoder:**
+   Leveraged the encoder part of the TrOCR model to obtain reliable feature representations from the images, focusing on capturing inherent characteristics of text. TrOCR encoder was used because, unlike CNNs the TrOCR feature representations contain textual details which would then be used to decode to characters. In essence, the encoder preserves textual information that CNNs might not.
```
Comment (mvoong18) [inline trocr/evaluation-dataset/handwritten-typed-text-classification/research.md:15]: This sentence is a bit unclear "TrOCR encoder was used because, unlike CNNs the TrOCR feature representations contain textual details which would then be used to decode to characters."

File: trocr/README.md [hunk 2]
```diff
@@ -65,32 +66,41 @@ cd ml-herbarium
...
-Create a new conda environment and activate it
+Create a new conda environment with the environment YAML file and activate it
 ```
-conda create -n my-conda-env python=3.9
+conda env create -n my-conda-env --file=trocr_env.yml
 conda activate my-conda-env
 ```
 
-Install all required packages and Jupter
+Install Jupter and required packages
```
Comment (mvoong18) [inline trocr/README.md:75]: change Jupeter to Jupyter

File: trocr/evaluation-dataset/handwritten-typed-text-classification/research.md (hunk 1 above)
Comment (kabilanmohanraj) [reply trocr/evaluation-dataset/handwritten-typed-text-classification/research.md]: I have rewritten this portion, Michelle.

File: trocr/README.md (hunk 2 above)
Comment (kabilanmohanraj) [reply trocr/README.md:75]: I have committed the correction

**Comment by kabilanmohanraj** [comment]
@mvoong18 Can you please postpone merging this branch until Prof. Thomas has reviewed this patch?

File: trocr/evaluation-dataset/handwritten-typed-text-classification/research.md
Comment (trgardos) [inline trocr/evaluation-dataset/handwritten-typed-text-classification/research.md:1]: Can you summarize the results of your handwritten/typed classifier?

Also, it would be helpful to explain a bit what the files are in this folder.

File: trocr/evaluation-dataset/handwritten-typed-text-classification/research.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# [Research] TrOCR Encoder + FFN Decoder
+
+#### Overview
+To create a robust classification model for our task, multiple Convolutional Neural Network (CNN) models were explored and assessed. Details of each attempted model, along with their respective implementations, can be accessed in the [Introduction section](https://github.com/BU-Spark/ml-herbarium/blob/dev/trocr/evaluation-dataset/handwritten-typed-text-classification/notebooks/Classifier_NN.ipynb) of the project's Jupyter Notebook.
```
Comment (trgardos) [inline trocr/evaluation-dataset/handwritten-typed-text-classification/research.md:4]: "for our task" -- There is no description of what the task is. I suggest starting the readme with the problem statement.

Comment (trgardos) [reply trocr/evaluation-dataset/handwritten-typed-text-classification/research.md:4]: Oh. I see there is a readme. Would be good to link the readme and explain what this doc is versus the readme. Presumably it is notes from your research this summer.

File: trocr/evaluation-dataset/handwritten-typed-text-classification/research.md
Comment (trgardos) [reply trocr/evaluation-dataset/handwritten-typed-text-classification/research.md:1]: Ok. I just noticed the readme.  But that doesn't have results either. So suggest putting them in one or the other and then cross linking between the 2 files.

Comment (trgardos) [reply trocr/evaluation-dataset/handwritten-typed-text-classification/research.md:1]: Btw, in the overview of the readme, it would be good to mention why you want to classify handwritten versus typed. At the very least I understood it was so that we can evaluate how the OCR does on handwritten versus typed.

File: trocr/label-extraction/research.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# [Research] DETR (DEtection TRansformer)
+
+#### Overview
+To choose an optimal model for detecting labels in plant sample images, a review of various models was undertaken. The task was to discern labels from plant specimen images, with potential models including LayoutLMv3 and DETR. A detailed comparison and a critical review of the models led to an optimal model selection, aligning with the project's specific goals and constraints.
```
Comment (trgardos) [inline trocr/label-extraction/research.md:4]: Put the actual links to the papers you cite, please.

File: trocr/label-extraction/research.md
Comment (trgardos) [inline trocr/label-extraction/research.md:1]: Here also, it would be good to post some results of your experiments. Either in this doc or in the readme.

File: trocr/docker/ReadMe.md
Comment (trgardos) [inline trocr/docker/ReadMe.md:1]: Great readme. Thanks!
